### DISCUSSION TITLE
Custom proxy route with vercel

### DISCUSSION THREAD

**Discussion by kochis** [vercel/vercel#8398] [Help] [answered]
Hi there,

We are currently using next.js with a custom express server, and are looking to move to Vercel. One issue is we have a custom proxy route for rendering documentation under the same domain (it's hosted on another site).

For instance, we have the following route in our server:
```js
    // proxy docs from the hosted github pages
    server.use('/documentation', createProxyMiddleware({
      target: 'https://<our-docsite>.github.io',
      changeOrigin: true,
      followRedirects: true,
    }));
```

I'm wondering if there's any way to support this behavior in Vercel, or if this needs to be handled at a higher level (like Cloudflare for instance).

Thanks!

**Comment by amyegan** [comment, marked as answer]
Hi @kochis. You can use [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) in a `vercel.json` [Project config](https://vercel.com/docs/project-configuration) for URL proxy. It is also possible to add a [Next.js rewrite](https://nextjs.org/docs/api-reference/next.config.js/rewrites) in `next.config.js` if you want to handle it there instead.  [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) is also an option.
